### PR TITLE
Flags

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Adds (better) plantable flags.
image

They can be deployed in place or simply carried around. 

They provide an aura buff that lowers your crit threshold (i.e. total health is the same but you can take an extra hit or 3 before falling over), as well as increased pain resistance.

Letting the flag touch the ground (or even worse, having someone from another faction take it) swaps the buff with a debuff, INCREASING pain and crit threshold.

TGMC flag by Fitz, SOM flag by myself.

Also a minor rework of how aura buff hud is set, to be less hot garbage. I need to do the same for beno though but its kinda out of scope.

Made some changes and fixes relating to deployables, a lot of it being old shitcode I'm responsible for.
## Why It's Good For The Game
Fun.
## Changelog
:cl:
add: Added deployable flags. Defend the colours!
code: Cleaned up some deployable component code
/:cl:

### CODE REVIEW

**Comment by Tonklaatech** [comment]
Me when my stolen ideas come to fruition

**Comment by FosterGrime-Heymynameslouie** [comment]
is this also for normal tgmc or just for hvh?
would be fun to have marine equiv of phero towers

**Comment by Lumipharon** [comment]
they'll be in hvx when its done

**Comment by Looking4bros** [comment]
THERE'S NO TIME TO FLOP AROUND LIKE A CARP, STAND AND FIGHT!!!!!!!!!!!!!!!!!!!

**Comment by Nerev4r** [comment]
Sin Kiske gameplay 2024

File: code/game/data_huds.dm [hunk 1]
```diff
@@ -620,30 +620,41 @@
...
-		if(protection_aura)
-			tempname += "hold"
-		if(marksman_aura)
-			tempname += "focus"
-		if(tempname)
-			holder.icon = 'icons/mob/hud/aura.dmi'
-			holder.icon_state = "[tempname]"
-
-	hud_list[ORDER_HUD] = holder
+	holder.overlays.Cut()
```
Comment (ivanmixo) [inline code/game/data_huds.dm:626]: I know it's not your fault but this should really be done properly (in another PR probably)

**Review by ivanmixo — APPROVED** [review]

File: code/game/data_huds.dm (hunk 1 above)
Comment (ivanmixo) [reply code/game/data_huds.dm:626]: Feel free to TM yourself

**Comment by zzurge** [comment]
Future standard bearer main here,DEATH TO THE DOGS OF TERRA

**Comment by Autisem** [comment]
maby add a way to "respectuflly put it down" so that when you literly just wana store it its not a huge problem, I dont care if its a 10 seconds do_after

**Comment by novaepee** [comment]
Would like if this was added to HvX; admin spawned the flag in HvX, and it was loved.

**Comment by lbnesquik** [comment]
Flag poles used to have pointy bits on the top. Please add one so I can stab people. Perhaps going on harm makes you lower it for stabbing.

**Comment by Lumipharon** [comment]
waiting on the acid pr to be merged first.

File: code/game/objects/items/plantable_flags.dm [hunk 2]
```diff
@@ -1,60 +1,225 @@
...
+	var/turf/origin_point
+
+/obj/item/plantable_flag/Initialize(mapload)
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_CAMPAIGN_MISSION_ENDED, PROC_REF(mission_end))
+	origin_point = get_turf(src)
+	if(isturf(loc))
+		item_flags |= DEPLOY_ON_INITIALIZE
+	AddComponent(/datum/component/deployable_item, /obj/structure/plantable_flag, 1 SECONDS, 3 SECONDS)
+	AddComponent(/datum/component/shield, SHIELD_PURE_BLOCKING, list(MELEE = 35, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, FIRE = 0, ACID = 0))
```
Comment (TiviPlus) [inline code/game/objects/items/plantable_flags.dm:45]: Shield?

File: code/game/objects/items/plantable_flags.dm [hunk 3]
```diff
@@ -1,60 +1,225 @@
...
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	worn_icon_list = list(
+		slot_l_hand_str = 'icons/mob/inhands/items/large_flag_left.dmi',
+		slot_r_hand_str = 'icons/mob/inhands/items/large_flag_right.dmi',
+	)
+	w_class = WEIGHT_CLASS_HUGE
+	force = 55
+	attack_speed = 15
+	attack_verb = list("stabbed", "thrust", "smashed", "thumped", "bashed", "attacked", "clubbed", "speared", "jabbed", "torn", "gored")
```
Comment (TiviPlus) [inline code/game/objects/items/plantable_flags.dm:26]: Torn? Isn't this grammatically incorrect

File: code/game/objects/items/plantable_flags.dm [hunk 4]
```diff
@@ -1,60 +1,225 @@
...
-		if(!do_after(usr, 1 SECONDS, NONE, src, BUSY_ICON_BUILD))
-			to_chat(user, "<span class='warning'>You decide against removing the flag here.")
-			return
-		is_collapsed = TRUE
-		update_appearance()
-	return ..()
+///Waves the flag around heroically
+/obj/item/plantable_flag/proc/lift_flag(mob/user)
+	if(TIMER_COOLDOWN_CHECK(user, COOLDOWN_WHISTLE_WARCRY))
+		to_chat(user, span_notice("You have to wait a while to rally your troops..."))
```
Comment (TiviPlus) [inline code/game/objects/items/plantable_flags.dm:119]: Balloooon

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Rest lgtm

File: code/game/objects/items/plantable_flags.dm (hunk 4 above)
Comment (Lumipharon) [reply code/game/objects/items/plantable_flags.dm:119]: nou. We don't need balloon alerts for everything, this is just an antispam message

File: code/game/objects/items/plantable_flags.dm (hunk 2 above)
Comment (Lumipharon) [reply code/game/objects/items/plantable_flags.dm:45]: ye? Its taking up a hand, and will debuff your whole team the instant you drop it, so it has some melee block chance to help them not be fodder

File: code/game/objects/items/plantable_flags.dm (hunk 3 above)
Comment (Lumipharon) [reply code/game/objects/items/plantable_flags.dm:26]: Isn't it? Its used by like every melee weapon. 
image

File: code/game/objects/items/plantable_flags.dm (hunk 4 above)
Comment (TiviPlus) [reply code/game/objects/items/plantable_flags.dm:119]: Exactly, it is something you only need to know in the immediate moment

File: code/game/objects/items/plantable_flags.dm (hunk 3 above)
Comment (TiviPlus) [reply code/game/objects/items/plantable_flags.dm:26]: shrug

File: code/game/objects/items/plantable_flags.dm (hunk 2 above)
Comment (TiviPlus) [reply code/game/objects/items/plantable_flags.dm:45]: ehh ok @Kuro020 fyi

File: code/game/objects/items/plantable_flags.dm (hunk 4 above)
Comment (Lumipharon) [reply code/game/objects/items/plantable_flags.dm:119]: balloon alert is way more stinky in this case but its changed.

**Comment by yyzsong** [comment]
Flags seem to bypass the self-harm restriction

**Pull request merged**